backend/src/main/java/de/qaware/mercury/storage/shop/impl/ShopDataRepository.java

```diff
@@ -10,23 +10,60 @@
 public interface ShopDataRepository extends JpaRepository<ShopEntity, UUID> {
     List<ShopEntity> findByName(String name);
 
+    /**
+     * Finds all approved and enabled shops
+     *
+     * @return all approved and enabled shops
+     */
     @Query(
-        "SELECT s as shopEntity, " +
-            "sqrt(" +
-            "  power(111.3 * cos((:latitude + s.latitude) / (2 * 0.01745)) * (:longitude - s.longitude) , 2)" +
-            " + power(111.3 * (:latitude - s.latitude), 2)" +
-            ") as distance FROM ShopEntity AS s " +
-            "WHERE s.enabled = true AND s.approved = true"
+        "SELECT s as shopEntity  FROM ShopEntity AS s WHERE s.enabled = true AND s.approved = true"
     )
-    List<ShopWithDistanceProjection> findNearby(@Param("latitude") double latitude, @Param("longitude") double longitude);
+    List<ShopEntity> findActive();
 
+    /**
+     * Finds all approved and enabled shops within the given coordinates matching the given query string
+     *
+     * @param query        a query string, supporting '%' as a 'like' operator
+     * @param maxLatitude  the maximum latitude
+     * @param maxLongitude the maximum longitude
+     * @param minLatitude  the minimum latitude
+     * @param minLongitude the minimum longitude
+     * @return all shop entities matching the given query within the given coordinates
+     */
     @Query(
-        "SELECT s as shopEntity, " +
-            "sqrt(" +
-            "  power(111.3 * cos((:latitude + s.latitude) / (2 * 0.01745)) * (:longitude - s.longitude) , 2)" +
-            " + power(111.3 * (:latitude - s.latitude), 2)" +
-            ") as distance FROM ShopEntity AS s " +
+        "SELECT s as shopEntity FROM ShopEntity AS s " +
```

Comment:
Siehe oben: `SELECT s FROM ShopEntity s ...`

Reply:
Ist `<` und `>` hier richtig? Damit ist min und max immer exklusiv.

Reply:
Angepasst. Hintergrund: in der Praxis macht das keinen Unterschied, da die Bounding Box > Radius ist, außer genau an den Punkten der Lotrechten. Die Parameter sind double Werte, in der DB mit 3-5 Nachkommastellen, bei uns in der Berechnung mit ~12 Nachkommastellen. Eine exakte Kollision an den Kanten ist also so wahrscheinlich wie ein Lottogewinn. Semantisch ist <= und >= richtig, also macht es zumindest vom Lesefluss her Sinn und > und < allein haben auch keinen Vorteil.